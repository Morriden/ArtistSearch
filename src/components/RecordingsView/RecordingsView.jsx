import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const RecordingsView = ({ recordings, bandName }) => {

  const recordingsArray = recordings.map(song => (
    <li key={song.id}>
      <Link to={`/lyrics/${bandName}/${song.title}`}>
        <div>
          {song.title}
        </div>
      </Link>
    </li>
  ));

  return (
    <ul>
      {recordingsArray}
    </ul>
  );
};
  
RecordingsView.propTypes = {
  recordings: PropTypes.arrayOf(PropTypes.object).isRequired,
  bandName: PropTypes.string.isRequired
};

export default RecordingsView;
