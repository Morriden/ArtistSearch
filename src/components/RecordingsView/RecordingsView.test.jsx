import React from 'react';
import { shallow } from 'enzyme';
import RecordingsView from './RecordingsView';


describe('Search Artists snap shot tests', () => {
  it('should be a list of artists', () => {
    const wrapper = shallow(<RecordingsView bandName={'im a band name'} recordings={[{
      title: 'title1', id: 'ID1'
    },
    {
      title: 'title2', id: 'ID2'
    }, {
      title: 'title3', id: 'ID3'
    }]}/>);
    
    expect(wrapper).toMatchSnapshot();
  });
});
