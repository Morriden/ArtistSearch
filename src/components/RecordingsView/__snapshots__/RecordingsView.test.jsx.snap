// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Artists snap shot tests should be a list of artists 1`] = `
<ul>
  <li
    key="ID1"
  >
    <Link
      to="/lyrics/im a band name/title1"
    >
      <div>
        title1
      </div>
    </Link>
  </li>
  <li
    key="ID2"
  >
    <Link
      to="/lyrics/im a band name/title2"
    >
      <div>
        title2
      </div>
    </Link>
  </li>
  <li
    key="ID3"
  >
    <Link
      to="/lyrics/im a band name/title3"
    >
      <div>
        title3
      </div>
    </Link>
  </li>
</ul>
`;
